modules/signals/src/ts-helpers.ts

```diff
@@ -1,13 +1,14 @@
 export type Prettify<T> = { [K in keyof T]: T[K] } & {};
 
 export type IsRecord<T> = T extends object
-  ? T extends unknown[]
-    ? false
-    : T extends Set<unknown>
-    ? false
-    : T extends Map<unknown, unknown>
-    ? false
-    : T extends Function
+  ? T extends
```

Comment:
Do want to extend this check (and within `isRecord`) to include all other types as well, including lesser known/used ones?

With help from Copilot:

```ts
type NonRecordTypes =
  | unknown[]
  | Set<unknown>
  | Map<any, unknown>
  | WeakSet<unknown>
  | WeakMap<any, unknown>
  | Date
  | Error
  | RegExp
  | Function
  | Promise<unknown>
  | ArrayBuffer
  | SharedArrayBuffer
  | DataView
  | Int8Array
  | Uint8Array
  | Uint8ClampedArray
  | Int16Array
  | Uint16Array
  | Int32Array
  | Uint32Array
  | Float32Array
  | Float64Array;

export type IsRecord<T> = T extends object
  ? T extends NonRecordTypes
    ? false
    : true
  : false;
```

Copilot also suggested this can be made simpler, but AFAIK there's no way of making `isRecord` more generic, instead of defining all possible types.

```ts
type IsRecord<T> = T extends object
  ? T extends Error
    ? false
    : keyof T extends string | number
    ? true
    : false
  : false;
```

Reply:
PR is updated with checks for additional built-in object types. Array, set, map, and uint/float arrays are covered with a check for `Iterable`.